### ISSUE TITLE
Docker DNS stops resolving

### ISSUE THREAD

**Issue by ahromis** [docker/docker#27868]
**Description**

The Docker DNS resolver stops resolving external DNS. 

AWS security groups were being changed (unsure about the extent of the changes) when this error started happening, and when the security group changes were reverted the DNS resolver still was not working. This issue was noticed when trying to login from DTR, and an OpenID Restarting the Docker daemon resolved the issue.

**Steps to reproduce the issue:**

I haven't been able to reproduce the issue myself, but it has happened a couple of times so far from DTR. This is my best guess on how to reproduce the issue.
1. Block DNS and HTTPS requests from DTR nodes with security groups or iptables
2. Try to login via DTR
3. Restore DNS and HTTPS egress traffic
4. Try to login again via DTR

**Describe the results you received:**

```
{
    "errors": [
{
    "code": "OPENID_ERROR",
    "message": "Failed to establish openid authentication",
    "detail": "OpenID Connect Error\n\nunable to get token\n\nsee server logs for details"
    }
  ]
}
```

DTR API log

```
{"level":"error","msg":"unable to get identity token: http client error: Post https://docker-ucp.domain.com/enzi/v0/id/token: dial tcp: lookup docker-ucp.domain.com on 127.0.0.11:53: read udp 127.0.0.1:56700-\u003e127.0.0.11:53: i/o timeout","time":"2016-10-24T21:12:51Z"}
```

Docker daemon log

```
[root@docker-ucp-node1 ~]# journalctl -fu docker.service
-- Logs begin at Thu 2016-05-05 22:40:27 UTC. --
Oct 27 14:39:52 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:39:52.510570551Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:40:36 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:40:36.801758741Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:41:54 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:41:54.424452500Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:42:33 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:42:33.119541626Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:43:34 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:43:34.919834843Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:44:06 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:44:06.002086422Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:45:11 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:45:11.453205348Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:46:13 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:46:13.731957348Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:46:24 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:46:24.778044301Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
Oct 27 14:47:35 docker-ucp-node1.domain.com docker[26726]: time="2016-10-27T14:47:35.627021835Z" level=warning msg="list flocker: TimeoutError: User timeout caused connection failure."
```

**Describe the results you expected:**

Expected that the DTR API container would resolve the UCP hostname, however it was not able to. Resolving DNS from the host machine worked without issue, as well as checking connectivity to port 443 against the UCP DNS address.

**Additional information you deem important (e.g. issue happens only occasionally):**

Asked the DTR team and they said DTR was behaving as expected and this appears to be a core/networking issue.

**Output of `docker version`:**

```
Client:
 Version:      1.11.2-cs3
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   c81a77d
 Built:        Wed Jun  8 01:23:22 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2-cs3
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   c81a77d
 Built:        Wed Jun  8 01:23:22 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 13
 Running: 11
 Paused: 0
 Stopped: 2
Images: 117
Server Version: 1.11.2-cs3
Storage Driver: devicemapper
 Pool Name: docker-202:1-562036864-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 72.21 GB
 Data Space Total: 107.4 GB
 Data Space Available: 35.16 GB
 Metadata Space Used: 82.22 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.065 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-12-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local flocker
 Network: overlay host bridge null
Kernel Version: 4.3.2-1.el7.elrepo.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.67 GiB
Name: docker-ucp-node1.domain.com
ID: GYIX:7H5T:NM7S:TA4W:ZW4Z:B37F:MCBW:S4CR:4HZ6:QCJN:OCAZ:3XIP
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-ip6tables is disabled
Cluster store: etcd://10.240.125.152:12379,10.240.251.114:12379,52.37.183.236:12379
Cluster advertise: 10.240.1.218:12376
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

AWS environment. 

Universal Control Plane: 1.1.2 (bbb5902)API: 1.22

DTR Version: 2.0.2

**Comment by sanimej** [comment]
@ahromis This looks like https://github.com/docker/docker/issues/22185. This has been fixed in 1.12. Can you give it a try with 1.12.3 ?

**Comment by ahromis** [comment]
Closing issue, I moved it to a different repo.
